watcher/watcher_test.go

```diff
@@ -16,292 +15,293 @@ func NewWatchHandlerMock() *WatchHandler {
-				Items: []core1.Pod{
-					{
-						ObjectMeta: v1.ObjectMeta{
-							Name:      "test",
-							Namespace: "default",
-						},
-						TypeMeta: v1.TypeMeta{
-							Kind: "pod",
-						},
-						Status: core1.PodStatus{
-							ContainerStatuses: []core1.ContainerStatus{
-								{
-									ImageID: "docker-pullable://alpine@sha256:1",
-									Name:    "container1",
-								},
-							},
-						},
-					},
-					{
-						ObjectMeta: v1.ObjectMeta{
-							Name:      "test2",
-							Namespace: "default",
-						},
-						TypeMeta: v1.TypeMeta{
-							Kind: "pod",
-						},
-						Status: core1.PodStatus{
-							ContainerStatuses: []core1.ContainerStatus{
-								{
-									ImageID: "docker-pullable://alpine@sha256:1",
-									Name:    "container2",
-								},
-							},
-						},
-					},
-				},
-			},
-			expectedImageIDsMap: map[string][]string{
-				"alpine@sha256:1": {pkgwlid.GetWLID("", "default", "pod", "test"), pkgwlid.GetWLID("", "default", "pod", "test2")},
-			},
-		},
-		{
-			name: "two wlids two image ids",
-			podList: core1.PodList{
-				Items: []core1.Pod{
-					{
-						ObjectMeta: v1.ObjectMeta{
-							Name:      "test",
-							Namespace: "default",
-						},
-						TypeMeta: v1.TypeMeta{
-							Kind: "pod",
-						},
-						Status: core1.PodStatus{
-							ContainerStatuses: []core1.ContainerStatus{
-								{
-									ImageID: "docker-pullable://alpine@sha256:1",
-									Name:    "container1",
-								},
-							},
-						},
-					},
-					{
-						ObjectMeta: v1.ObjectMeta{
-							Name:      "test2",
-							Namespace: "default",
-						},
-						TypeMeta: v1.TypeMeta{
-							Kind: "pod",
-						},
-						Status: core1.PodStatus{
-							ContainerStatuses: []core1.ContainerStatus{
-								{
-									ImageID: "docker-pullable://alpine@sha256:2",
-									Name:    "container2",
-								},
-							},
-						},
-					}}},
-			expectedImageIDsMap: map[string][]string{
-				"alpine@sha256:1": {pkgwlid.GetWLID("", "default", "pod", "test")},
-				"alpine@sha256:2": {pkgwlid.GetWLID("", "default", "pod", "test2")},
-			},
-		},
-		{
-			name: "one wlid two image ids",
-			podList: core1.PodList{
-				Items: []core1.Pod{
-					{
-						ObjectMeta: v1.ObjectMeta{
-							Name:      "test",
-							Namespace: "default",
-						},
-						TypeMeta: v1.TypeMeta{
-							Kind: "pod",
-						},
-						Status: core1.PodStatus{
-							ContainerStatuses: []core1.ContainerStatus{
-								{
-									ImageID: "docker-pullable://alpine@sha256:1",
-									Name:    "container1",
-								},
-								{
-									ImageID: "docker-pullable://alpine@sha256:2",
-									Name:    "container2",
-								},
-							},
-						},
-					}}},
-			expectedImageIDsMap: map[string][]string{
-				"alpine@sha256:1": {pkgwlid.GetWLID("", "default", "pod", "test")},
-				"alpine@sha256:2": {pkgwlid.GetWLID("", "default", "pod", "test")},
-			},
-		},
-	}
+// func TestBuildImageIDsToWlidsMap(t *testing.T) {
```

Comment:
So according to @dwertent review I added an internal map of `instanceIDs`. This caused me to use K8S API inside the buildMap function, which makes me enable to unit test it. I put it on comment since there is a chance this map will be removed, and then I can return to the old function and its unit tests

Reply:
ok good